recommendations/index.md

```diff
@@ -149,3 +149,53 @@ Although JSON API does not specify the format of date and time fields, it is
 recommended that servers align with ISO 8601. [This W3C
 NOTE](http://www.w3.org/TR/NOTE-datetime) provides an overview of the
 recommended formats.
+=======
+
+## Asynchronous Processing <a href="#asynchronous-processing" id="asynchronous-processing" class="headerlink"></a>
+
+Consider a situation when you need to create a resource and the operation takes long time to complete.
+
+The request **SHOULD** return a status `202 Accepted` with a link in Header/Location.
+
+```text
+HTTP/1.1 202 Accepted
+Location: https://example.com/photos/queue/5234
```

Comment:
Yes, a link in `links` could be enough, but I don't think a top-level `self` link is the right link. The top-level `"self"` is defined as the URI the request was made to. So, the top-level `self` (if present at all—and, in fact, it probably doesn't make sense to include in response to a POST) would be the URI the user sent the job to, not the URI of the queued job itself.

A resource-level `self` link could work, though. So the request would be `POST /tasks` and the response would be:

```text
HTTP/1.1 202 Accepted
Content-Type: application/vnd.api+json
```
```js
{
   "data": {
      "type": "queued",
      "id": 1234",
      "attributes": { "status": "pending..." },
      "links": { "self": "/photos/queue/5234"} // <- the link would go here
   }
```

Does that work for you?

It seems to very nicely comply with HTTP, which says that:

> The representation sent with [a 202] response ought to describe the request's current status and point to (or embed) a status monitor that can provide the user with an estimate of when the request will be fulfilled.